processing/src/main/java/org/apache/druid/query/groupby/strategy/GroupByStrategyV2.java

```diff
@@ -419,36 +436,81 @@ public Row apply(final Row row)
                         .collect(Collectors.toList())
         );
 
-        subtotalsResults.add(applyPostProcessing(
-            mergeResults(new QueryRunner<Row>()
-            {
-              @Override
-              public Sequence<Row> run(QueryPlus<Row> queryPlus, Map<String, Object> responseContext)
+        if (Utils.isPrefix(subtotalSpec, queryDimNames)) {
+          subtotalsResults.add(applyPostProcessing(
+              mergeResults(new QueryRunner<Row>()
               {
-                return GroupByRowProcessor.getRowsFromGrouper(
-                    queryWithoutSubtotalsSpec,
-                    subtotalSpec,
-                    grouperSupplier
-                );
-              }
-            }, subtotalQuery, null),
-            subtotalQuery
-                             )
-        );
+                @Override
+                public Sequence<Row> run(QueryPlus<Row> queryPlus, Map<String, Object> responseContext)
+                {
+                  return GroupByRowProcessor.getRowsFromGrouper(
+                      queryWithoutSubtotalsSpec,
+                      subtotalSpec,
+                      grouperSupplier
+                  );
+                }
+              }, subtotalQuery, null),
+              subtotalQuery
+                               )
+          );
+        } else {
+          subtotalsResults.add(applyPostProcessing(
+              mergeResults(new QueryRunner<Row>()
+              {
+                @Override
+                public Sequence<Row> run(QueryPlus<Row> queryPlus, Map<String, Object> responseContext)
+                {
+                  List<Closeable> closeables = new ArrayList<>();
+
+                  Sequence<Row> result = GroupByRowProcessor.getRowsFromGrouper(
+                      subtotalQuery,
+                      subtotalSpec,
+                      Suppliers.memoize(() -> GroupByRowProcessor.createGrouper(
+                          subtotalQuery,
+                          GroupByRowProcessor.getRowsFromGrouper(
+                              queryWithoutSubtotalsSpec,
+                              subtotalSpec,
+                              grouperSupplier
+                          ),
+                          GroupByQueryHelper.rowSignatureFor(subtotalQuery),
+                          configSupplier.get(),
+                          resource,
+                          spillMapper,
+                          processingConfig.getTmpDir(),
+                          processingConfig.intermediateComputeSizeBytes(),
+                          closeables,
+                          false,
+                          false
+                                        )
+                      )
+                  );
+
+                  return Sequences.withBaggage(result, () -> Lists.reverse(closeables).forEach(closeable -> CloseQuietly.close(closeable)));
+                }
+              }, subtotalQuery, null),
```

Comment:
Something looks weird about this formatting…

Reply:
it was passing the intellij formatter , introduced few newlines and it looks better now.